Atome

BaQs opened the pull request:
## Breaking Change:

<!-- What is breaking and why we have to break it. Remove this section only if it was NOT a breaking change. -->

## Description:
The `atome` sensor platform is retrieving the consumption of your home from the [Direct Energy Atome electric meter](https://total.direct-energie.com/particuliers/electricite/compteur-linky/atome).
This special little device is connected to a Linky Electric Meter, and sends live data to a cloud platform.


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>
10172: https://github.com/home-assistant/home-assistant.io/pull/10172


## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: atome
    username: YOUR_ATOME_USERNAME
    password: YOUR_ATOME_PASSWORD
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

balloob committed:
Bump version to 0.97.0b0

dmulcahey committed:
bump quirks (#25618)

Kane610 committed:
Handle disabled devices (#25625)

Jc2k committed:
Fix polling HomeKit devices with multiple services per accessory (#25629)

balloob committed:
Filter out empty results in history API (#25633)

balloob committed:
Expose comfort presets as HA presets (#25491)

* Expose comfort presets as HA presets

* Fix bugs

* Handle unavailable

* log level debug on update

* Lint

Jc2k committed:
Bump homekit_python to 0.15 (#25631)

oncleben31 committed:
Meteofrance improve log error messages (#25630)

* Improve log error messages

* remove unique_id not ready yet

DarkFox committed:
Fix handling of empty results from Rejseplanen (#25610)

* Improve handling of empty results from Rejseplanen (Fixes #25566)

* Exclude attributes with null value

* Add period back into docstring

* Fix formatting

balloob committed:
Upgrade hass-nabucasa to 0.16 (#25636)

balloob committed:
Updated frontend to 20190801.0

balloob committed:
Bumped version to 0.97.0b1

balloob committed:
Add preset to be away and eco (#25643)

Commit:
Add HmIP-SCI to Homematic IP Cloud, Fix HmIP-SWDM (#25639)

* Add HmIP-SCI to Homematic IP Cloud

* Bump upstream dependency

* Fix HmIP-SWDM

Santobert committed:
Feature zwave preset modes (#25537)

* Initial commit

* Add some more code

* Local tests passing

* Remove unnecessary line

* Add preset attributes to __init__

* Remove some more debugger lines

* Add some tests

* Fix comparision to None

* Improve test coverage

* Use unknown modes as presets

* Bugfixes and test improvements

* Add tests for unknown preset modes

* linting

* Improve mappings

* Move PRESET_MANUFACTURER_SPECIFIC to zwave

* Replace isinstance with cast

* Add test for hvac_action

* hvac_mode is never None

* Improved mapping of current mode to hvac/preset modes

* Fix bugs where hvac_mode is None

* Add default hvac mode

* Fixed default hvac mode

* Fix linting

* Make flake happy

* Another linting

* Make black happy

* Complete list of default hvac modes

* Add mapping to heat/cool eco

* Fixed another bug where mapping goes wrong

Kane610 committed:
UniFi - allow configuration to not track clients or devices (#25642)

* Allow configuration to not track clients or devices

amelchio committed:
Revert flux_led to 0.89 (#25653)

* Revert Black

* Revert "Introduce support for color temperature (#25503)"

This reverts commit e1d884a4841f853bb4247a76fca417c0b4f4e7c1.

* Revert "Fix flux_led only-white controllers (#22210)"

This reverts commit 48138189b3c24261fe62a78b6ec854c761d5ce63.

* Revert "Fix MagicHome LEDs with flux_led component (#20733)"

This reverts commit 1444a684e02fab99648f6e5daea9f28b6cf45c10.

* Re-Black

* Use mode detection for scanned bulbs

Kane610 committed:
Options to not track wired clients (#25669)

Commit:
Add PRESET_AWAY to HomematicIP Cloud climate (#25641)

* enable climate away_mode and home.refresh

* Add Party eco modes

Kane610 committed:
In some circumstances device.last_seen can be None (#25690)

Kane610 committed:
UniFi - reverse connectivity logic (#25691)

* Make connectivity control in line with other implementations

bachya committed:
Fix issue with incorrect Notion bridge IDs (#25683)

* Fix issue with incorrect Notion bridge IDs

* Less aggressive

* Member comments

pvizeli committed:
Fix roku lxml requirement (#25696)

balloob committed:
Updated frontend to 20190804.0

balloob committed:
Bumped version to 0.97.0b2

balloob committed:
Update HTTP defaults (#25702)

* Update HTTP defaults

* Fix tests

balloob committed:
Add service to reload scenes from configuration.yaml (#25680)

* Allow reloading scenes

* Update requirements

* address comments

* fix typing

* fix tests

* Update homeassistant/components/homeassistant/scene.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Address comments

jesserizzo committed:
Bump envoy_reader to 0.8.6, fix missing dependency  (#25679)

* Bump envoy_reader to 0.8.6, fix missing dependency

* Bump envoy_reader to 0.8.6, fix missing dependency

Kane610 committed:
UniFi - handle device not having a name (#25713)

* Handle device not having a name

balloob committed:
Updated frontend to 20190805.0

balloob committed:
Bumped version to 0.97.0b3

zxdavb committed:
initial commit (#25731)

tsvi committed:
Bump hdate==0.9.0 (use pytz instead of dateutil) (#25726)

Use new hdate version of library which uses pytz for timezones.
dateutil expects /usr/share/timezone files, as these are not available
in the docker image and in HASSIO, the timezone offsets are broken.

This should fix
 - #23032
 - #18731

balloob committed:
Revert emulated hue changes (#25732)

Kane610 committed:
Fix last seen not available on certain devices (#25735)

balloob committed:
Bumped version to 0.97.0

balloob committed:
Update requirements

balloob committed:
Merge pull request #25756 from home-assistant/rc

0.97.0

jjlawren committed:
Don't track unstable attributes (#25787)

fbradyirl committed:
Update Cisco Mobility Express module version  (#25770)

* Update manifest.json

* Update requirements_all.txt

Anonym-tsk committed:
Fix deconz allow_clip_sensor and allow_deconz_groups options (#25811)

aetaric committed:
Update HIBP sensor to use API v3 and API Key (#25699)

* Update HIBP sensor to use API v3 and API Key

* ran black code formatter

* fixed stray , that was invalid in multiple json formatters

balloob committed:
Bumped version to 0.97.1

andrewsayre committed:
Fix brightness type (#25818)

Santobert committed:
Add script to install locale (#25791)

balloob committed:
Merge pull request #25819 from home-assistant/rc

0.97.1

brandond committed:
Update pyvera to 0.3.3 (#25820)

Fixes #24987

cameronrmorris committed:
Fix eco preset for Wink Air Conditioner (#25763)

* Add preset support for device
* Provide mappings between preset changes

cgtobi committed:
Fix Netatmo climate issue (#25830)

* Bump pyatmo to v2.2.1

* Fix issue 25778

tombbo committed:
Fix KNX Climate mode change callback (#25851)

- fix KNX Climate not updating UI after receiving mode change telegram
  from KNX bus

andrewsayre committed:
Always populate hvac_modes in SmartThings climate platform (#25859)

* Always return list for hvac_modes

* Use climate constants

Kane610 committed:
UniFi - Use state to know if device is online (#25876)

pvizeli committed:
Fix issue with nuki new available state (#25881)

balloob committed:
Bumped version to 0.97.2

balloob committed:
Merge pull request #25884 from home-assistant/rc

0.97.2

BaQs committed:
first commit of atome component

BaQs committed:
first commit of atome component

BaQs committed:
first commit of atome component

BaQs committed:
first commit of atome component

BaQs committed:
sanatizing + fixed some errors handling

BaQs committed:
sanatizing + fixed some errors handling

BaQs committed:
sanatizing + fixed some errors handling

BaQs committed:
fixed re-login

BaQs committed:
fixed re-login

BaQs committed:
fixed re-login

BaQs committed:
updated coverage for atom which relies on external API

BaQs committed:
updated coverage for atom which relies on external API

BaQs committed:
updated coverage for atom which relies on external API

BaQs committed:
Proper manifest for atome

BaQs committed:
Proper manifest for atome

BaQs committed:
Proper manifest for atome

BaQs committed:
Added default power class for Atome

BaQs committed:
Added default power class for Atome

BaQs committed:
Added default power class for Atome

BaQs committed:
flake8 rules are now respected

BaQs committed:
flake8 rules are now respected

BaQs committed:
flake8 rules are now respected

BaQs commented:
Please, as this is my first pull request, do not hesitate to propose any good practice I did not respect.

BaQs committed:
some pylint fixes

BaQs committed:
some pylint fixes

BaQs committed:
some pylint fixes

BaQs committed:
Again, some flake8 fixes...

BaQs committed:
Again, some flake8 fixes...

BaQs committed:
Again, some flake8 fixes...

ThaSiouL committed:
Fritz device tracker: set 'scanning' log to debug (#26043)

Changed the logging level for the polling of the fritz device tracker from info to debug.  The message was logged up to 6 times per minute and bloated the log file.

slackr31337 committed:
Fix Mikrotik ARP ping (#25965)

* Reuse ssl_wraper

* Fix arp_ping

* Restore debug log

* Fix attributes

BaQs committed:
First line should be in imperative mood

BaQs committed:
First line should be in imperative mood

BaQs committed:
First line should be in imperative mood

BaQs committed:
flake8 stupidity...

BaQs committed:
flake8 stupidity...

BaQs committed:
flake8 stupidity...

BaQs committed:
Flake8 fixes

BaQs committed:
Flake8 fixes

BaQs committed:
Flake8 fixes

BaQs committed:
Just a test

BaQs committed:
Just a test

BaQs committed:
Just a test

balloob committed:
Make sure config flows extend ConfigFlow base class (#26051)

MartinHjelmare commented on homeassistant/components/atome/sensor.py:
```diff
@@ -0,0 +1,217 @@
+        # Login the user into the Atome API.
+        payload = {"email": username, "plainPassword": password}
+
+        req = requests.post(
```
Please extract the device specific interface code and data parsing to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#4-communication-with-devices-services

BaQs replied:
Hi, thx I did that.

pvizeli committed:
Revert "Fix bmw_connected_drive and eq3btsmart components by updating their dependencies   (#26012)" (#26065)

This reverts commit 9e5243929e3821daa44ecb592406fa4a4da88cd1.

pvizeli committed:
Bump nabucasa-cloud to 0.17 (#26066)

fredrike committed:
fix alarm webhooks (#26062)

dannerph committed:
Add Keba charging station/wallbox as component (#24484)

* Add Keba charging station wallbox component

* Added start/stop commands (ena 0 and ena 1)

* added refresh_interval parameter and fixed authorization

* fixed max line length

* deactivate failsafe mode if not set in configuration

* extracted I/O code to pypi library

* updated services.yaml

* pinned version of requirements

* fixed typos, indent and comments

* simplified sensor generation, fixed unique_id and name of sensors

* cleaned up data extraction

* flake8 fixes

* added fast polling, fixed unique_id, code cleanup

* updated requirements

* fixed pylint

* integrated code styling suggestions

* fixed pylint

* code style changes according to suggestions and pylint fixes

* formatted with black

* clarefied variables

* Update homeassistant/components/keba/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Update homeassistant/components/keba/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Update homeassistant/components/keba/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Update homeassistant/components/keba/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* fixed behaviour if no charging station was found

* fix pylint

* Update homeassistant/components/keba/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

pvizeli committed:
Add packages version to Tensoflow for wheels packages (#26068)

* Add packages version to Tensoflow for wheels packages

* Uncomment tensorflow

* fix string

* Revert version

jkeljo committed:
Remove reference to typing.Deque (added in Python 3.6.1) (#26030)

* Remove reference to typing.Deque (added in Python 3.6.1)

* Silence mypy

* Type as collections.deque

flebourse committed:
huawei_lte: support out of range values in default sensor formatter (#26052)

* Update sensor.py

Change regexp to cope with out of range values, ie "rssi" : ">=-51dBm".

* Add tests for format_default

* Fix black formatting

kellerza committed:
SMA simplify config (#25880)

Kane610 committed:
Add config entry options support to deCONZ (#26049)

kellerza committed:
pysma library update 0.3.4 (#26075)

balloob committed:
Save config entry after updating system options (#26077)

balloob committed:
Updated frontend to 20190820.0

balloob committed:
Convert progress API to WS (#26082)

tkislan committed:
Add Minio component (#23567)

* Add minio implementation

* Static check changes

* Added docstrings

* Update docstrings

* Update docstrings

* Fix linter errors

* Finally fix all docstring errors

* Create services.yaml

* Update CODEOWNERS

* Final changes

* Remove double underscores

* Minor changes

* Update config.yml

* Review changes

* Added tests

* Fix lint errors

* Move tests from unittest to pytest

* Add minio as test requirement

* Update test_minio_helper.py

* Better event thread handling, added hass test

* Update tests

* Fixed lint errors

* Update test_minio.py

* Review changes

* More review changes

* Removed tests

* Applied code style changes

* Reformat test code

cj-thornton committed:
Add path option to SABnzbd component (#25908)

* Add path option to SABnzbd component

Adds an optional `path` setting to the SABnzbd component. This allows support for SABnzbd installs that use a different `url_base` (typically a reverse proxied configuration; see https://sabnzbd.org/wiki/configuration/2.3/special).

This change passes the `path` along to pysabnzbd as its `web_root`, which in turn uses the path to build up it's api URLs.

* Use dict.get for Sabnzbd web_root path config

pvizeli committed:
Update devcontainer.json

balloob committed:
Add strings for traccar config flow (#26089)

* Add strings for traccar config flow

* Update strings.json

konikvranik committed:
Update pyrainbird to version 0.2.1 to fix zone number (#26064)

* Update pyrainbird to version 0.2.0 to fix zone number issue:

- home-assistant/home-assistant/issues/24519
- jbarrancos/pyrainbird/issues/5
- https://community.home-assistant.io/t/rainbird-zone-switches-5-8-dont-correspond/104705

* requirements_all.txt regenerated

* code formatting

* code formatting

* response checking

* fixed switch state

* pyrainbird version bump

* formatting

* version bump

* if instead elif

legacycode committed:
Update pylacrosse library to version 0.4.0 (#26088)

* Updated pylacrosse library to version 0.4.0. Adding support for remote serial port.

* Generated new requirements_all.txt file.

pvizeli committed:
Docker ADR (#26085)

* Migrate Docker image to Hass.io / Multiarch

* Fix sudo

* Update CODEOWNERS

* Fix manifest

* Add more logic

* fix handling

* Move dockerfile

* Modify options

zxdavb committed:
Bump geniushub client (#26084)

* bump geniushub client

* delint

* remove unsused lint hints

balloob committed:
Use init_subclass for Config Entries (#26059)

* Use init_subclass for Config Entries

* Ignore type

ahertz committed:
Sonos playlists play media (#26054)

* Add handling for Sonos playlists in media_player.play_media

* Avoid breaking change by falling back to previous behavior

* Use the proper MEDIA_TYPE_PLAYLIST constant.

* Addressed comments, restricting media_type to music or playlist

ludeeus committed:
Remove the googlehome integration (#26035)

scop committed:
Test with 3.6.0 in Travis (#26039)

https://github.com/home-assistant/home-assistant/pull/26030#issuecomment-522298190

sander76 committed:
add id to state attributes (#26086)

balloob committed:
Fix open-ui cli arg (#26091)

* Fix open-ui cli command

* Align add_job typing with async_add_job

wolph committed:
[bugfix] Exception if vendor's servers are currently unavailable (#26093)

Adminiuga committed:
Bump up zha dependencies. (#26097)

balloob committed:
Updated frontend to 20190820.1

gerard33 committed:
Update bimmer_connected to 0.6.0 (#26098)

* Update bimmer_connected to 0.6.0

* Correct file properties

SukramJ committed:
Add support for warning attributes to Homematic IP Cloud (#26103)

* add supported optional features

* use recommendations

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

croghostrider committed:
check if a light supports brightness (#26055)

Fix black

Fix black

dmulcahey committed:
bump quirks (#26106)

BKPepe committed:
Upgrade ruamel_yaml to 0.15.100 (#26095)

scop committed:
Upgrade pydocstyle to 4.0.1 (#26111)

https://github.com/PyCQA/pydocstyle/blob/4.0.1/docs/release_notes.rst#401---august-14th-2019

scop committed:
Upgrade pytest to 5.1.1 (#26112)

https://docs.pytest.org/en/latest/changelog.html#pytest-5-1-0-2019-08-15
https://docs.pytest.org/en/latest/changelog.html#pytest-5-1-1-2019-08-20

scop committed:
Remove coveralls test dependency (#26110)

Outdated and unused.

scop committed:
Check and fix test suite leaving files behind (#25981)

* azure: run check_dirty at end of tests

* Fix ps4 media player tests to not write to files

* .gitignore coverage.xml and test-results.xml

TheLastProject committed:
Update PyEssent (#26115)

SukramJ committed:
remove ATTR_ID for groups (#26114)

amelchio committed:
Update eternalegypt to 0.0.10 (#26117)

Kane610 committed:
UniFi config entry options (#26113)

Introduce config entry options for Unifi integration
Allow configuration.yaml options to be imported to new options

gadgetchnnel committed:
Template binary sensor attributes (#22664)

* Added attribute support to template binary sensor with tests

Added attribute support to template binary sensor with tests

* fix dictionary update

fix dictionary update

* Fixed whitespace and line length issues

* Fixed indentation

* Simplify applying of attribute templates based on feedback

* Syntax and whitespace fixes

* Black formatting

* Black formatting on tests

* Check attribute_templates is not None

* Fixed test

* Added test for failure to render template

* Test fix

* Updated test

* Removed whitespace and applied Black formatting

* Fixed test assertion

* Updated test

* Code improvements folloing comments

Using chain to iterate over templates and attribute_templates
Replacing dict() with {}
Rmoving unused constant

* Applied Black formatting

* Fixed removed code

* Default attribute_templates to empty dict

* Black formatting

* Fixed imports

JeffLIrion committed:
Add descriptive fields to script config (#26056)

* Add descriptive fields to script config

* Add script descriptions to hass.data["service_description_cache"]

* Import SERVICE_DESCRIPTION_CACHE

* Register script descriptions via async_set_service_schema

* Add scripts test for loading and reloading service descriptions

* Minor cleanup

* Clean up script schema

balloob committed:
Updated frontend to 20190821.0

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Export external calls to PyAtome pypi package

BaQs committed:
Wrong get data function

BaQs committed:
Wrong get data function

BaQs committed:
Wrong get data function

BaQs committed:
Do not deserver yet to be code owner

BaQs committed:
Do not deserver yet to be code owner

BaQs committed:
Do not deserver yet to be code owner

balloob committed:
Fix tests leaving files behind (#26121)

frenck committed:
Statistics sensors repect given name (#26119)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines (#26128)

* Update azure-pipelines-release.yml for Azure Pipelines

* Update azure-pipelines-release.yml

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

SukramJ committed:
Add myself as codeowner to HmIP Cloud (#26140)

SukramJ committed:
Splitt device_state_attributes between device and group for Homematic IP Cloud (#26137)

* splitt device_state_attributes between device and group

* readd device_state_attributes for access point

kellerza committed:
DuckDNS setup backoff (#25899)

BaQs committed:
Added proper codeowner

BaQs committed:
Added proper codeowner

BaQs committed:
Added proper codeowner

JeffLIrion committed:
Load user-provided descriptions for python_scripts (#26069)

* Load user-provided descriptions for python_scripts

* Import SERVICE_DESCRIPTION_CACHE

* Use async_set_service_schema to register service descriptions

* Add python_script tests for loading service descriptions

* Use async/await in test

PaulAnnekov committed:
Fix tuya switch state (#26145)

* bump tuyaha 0.0.3

* bump tuyaha 0.0.3

filcole committed:
Nissanleaf login fix (#26139)

* Upgrade to pycarwings2.9 per 25 July 2019 API change

* Remove rest of location tracker. Fix get_status_from_update call.

BaQs committed:
Added pyatome requirement for Atome new platform

BaQs committed:
Added pyatome requirement for Atome new platform

BaQs committed:
Added pyatome requirement for Atome new platform

balloob commented on .coveragerc:
```diff
@@ -50,6 +50,7 @@ omit =
     homeassistant/components/arwn/sensor.py
     homeassistant/components/asterisk_cdr/mailbox.py
     homeassistant/components/asterisk_mbox/*
+    homeassistant/components/atome/sensor.py
```
```suggestion
    homeassistant/components/atome/*
```

BaQs committed:
Black reformatted

BaQs committed:
Black reformatted

BaQs committed:
Black reformatted

BaQs committed:
Update .coveragerc

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

BaQs committed:
Update .coveragerc

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

BaQs committed:
Update .coveragerc

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

balloob commented on homeassistant/components/atome/sensor.py:
```diff
@@ -0,0 +1,105 @@
+    try:
+        client = AtomeClient(username, password)
+    except Exception as exp:
+        _LOGGER.error(exp)
```
You should return here.

balloob commented on homeassistant/components/atome/sensor.py:
```diff
@@ -0,0 +1,105 @@
+    """Initiate Atome Client object"""
+    try:
+        client = AtomeClient(username, password)
+    except Exception as exp:
```
Don't catch generic exceptions.

balloob committed:
Log warning if disabled entities receive updates. (#26143)

* Log warning if disabled entities receive updates.

* Fix test

* Always set entity ID on disabled entities

Commit:
Logging

Commit:
Logging

Commit:
New pyAtome version

Commit:
Resolved conflicts

balloob committed:
Updated frontend to 20190822.0

Commit:
PyAtome 0.0.11 is now necessary

balloob committed:
Reload config entry when entity enabled in entity registry, remove entity if disabled. (#26120)

* Reload config entry when disabled_by updated in entity registry

* Add types

* Remove entities that get disabled

* Remove unnecessary domain checks.

* Attach handler in async_setup

* Remove unused var

* Type

* Fix test

* Fix tests

iamtpage committed:
Venstar: define success for all branches of set_temperature() (#26148)

* define success for all branches

* add operation_mode to error when unexpected value

* fix black linting

* black linting

* fix weird black linting result

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

MatthewFlamm committed:
Add NWS weather (#23647)

* Add nws weather.

* Hassfest

* Address multiple comments

* Add NWS icon weather code link

* Add metar fallback.

Use metar code from nws observation if normal api is missing data.

* only get 1 observation - we dont use more than 1

* add mocked metar for tests

* lint

* mock metar package for all tests

* add check for metar attributes

* catch errors in setup

* add timeout error

* handle request exceptions

* check and test for missing observations

* refactor to new pynws

* change to simpler api

* Make py3.5 compatible

Remove f string

* bump pynws version

* gen_requirements

* fix wind bearing observation

* Revert "Make py3.5 compatible"

This reverts commit 4946d91779a6e539ea43e667b2265557a49a0bb5.

* Precommit black missed a file?

* black test

* add exceptional weather condition

* bump pynws version

* update requirements_all

* address comments

* move observation and forecast outside try-except-else

* Revert "move observation and forecast outside try-except-else"

This reverts commit 53b78b32837b55b8a0b61de6192e846f6a486754.

* remove else from update forecast block

* remove unneeded ConfigEntryNotReady import

* add scan_interval, reduce min_time_between_updates

* pytest tests

* lint test docstring

* use async await

* lat and lon inclusive in config

JeffLIrion committed:
Bump androidtv to 0.0.24 (#26158)

* Bump androidtv to 0.0.24

* Add unique ID for Fire TV (not just Android TV)

OnFreund committed:
CoolMaster: Change auto to heat_cool (#26144)

frenck committed:
Use literal string interpolation in core (f-strings) (#26166)

chaoranxie committed:
fix issue setting scan_interval (#26165)

I was getting the following error when i set the scan_interval
```
    self.scan_interval = timedelta(seconds=config.get(CONF_SCAN_INTERVAL, 60))
TypeError: unsupported type for timedelta seconds component: datetime.timedelta
```
it turns out `config.get(CONF_SCAN_INTERVAL)` already returns `timedelta`

```('scan_interval', datetime.timedelta(seconds=180))```

balloob committed:
Allow bumping version for nightly builds (#26167)

Danielhiversen committed:
Update ambiclimate library (#26182)

Commit:
Proper exception handling, now need pyatome 0.0.15

Commit:
Proper exception handling, now need pyatome 0.0.15

Commit:
Requirement

Commit:
Requirement

MartinHjelmare commented:
Please clean up the branch and remove the unrelated commits.

bachya committed:
Fix possible KeyError in SimpliSafe (#26190)

Danielhiversen committed:
Refactor open garage (#26179)

* Use defined constant in open garage

* refactor open garage

* style

* style

* refactor

* refactor

* refactor

* remove vehicle state

BaQs commented:
> 
> 
> Please clean up the branch and remove the unrelated commits.

You mean to "rewrite the history" and make (for instance) one commit for the new platform ?

MartinHjelmare commented:
You can do an interactive `git rebase`. If you're not comfortable with `git` it might be easier to start over in a new branch and PR.

Commit:
Merge branch 'atome' of https://github.com/baqs/home-assistant into atome

BaQs commented:
> 
> 
> You can do an interactive `git rebase`. If you're not comfortable with `git` it might be easier to start over in a new branch and PR.

I just tried rebasing, I don't think this is better, right ?
( if it's the case I'll do a new PR indeed, sorry for that)

MartinHjelmare commented:
Just `git rebase` won't do the trick. You need to specify the commits to remove or keep somehow. One way of doing this is with the `-i` switch, to enter interactive mode. But please read up on git rebase before doing this.

BaQs commented:
Ok here is the clean PR:
https://github.com/home-assistant/home-assistant/pull/26194

BaQs commented:
Hmm, seems the codecov/patch does not pass on my other PR. If you have any recommendation, I would be very very glad!
